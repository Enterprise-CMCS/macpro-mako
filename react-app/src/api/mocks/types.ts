import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

type GetTypesBody = { authorityId: number };
type GetSubTypesBody = { authorityId: number; typeIds: number[] };

export const handlers = [
  http.post<any, GetTypesBody>("/os/getTypes", async ({ request }) => {
    const { authorityId } = await request.json();

    if (authorityId === -1) {
      throw Error("useGetTypes > mockFetch: Expected error thrown by test.");
    }

    return HttpResponse.json({
      hits: {
        hits: mockData.filter(
          (item) => item._source.authorityId === authorityId && !item._source.typeId,
        ),
      },
    });
  }),

  http.post<any, GetSubTypesBody>("/os/getSubTypes", async ({ request }) => {
    const { authorityId, typeIds } = await request.json();

    if (authorityId === -1) {
      throw Error("useGetSubTypes > mockFetch: Expected error thrown by test.");
    }

    const filteredData = mockData.filter(
      (item) =>
        item._source.authorityId === authorityId &&
        typeIds.some((typeId) => item._source.typeId === typeId),
    );

    return HttpResponse.json({
      hits: {
        hits: filteredData,
      },
    });
  }),
];

const mockData = [
  { _source: { id: 101, authorityId: 1, name: "typeOne" } },
  { _source: { id: 102, authorityId: 1, name: "typetwo" } },
  { _source: { id: 103, authorityId: 2, name: "typethree" } },
  { _source: { id: 101, authorityId: 1, name: "subtypeOne", typeId: 1 } },
  { _source: { id: 102, authorityId: 1, name: "subtypetwo", typeId: 2 } },
  { _source: { id: 103, authorityId: 2, name: "subtypethree", typeId: 1 } },
  { _source: { id: 104, authorityId: 2, name: "subtypethree", typeId: 4 } },
  { _source: { id: 105, authorityId: 2, name: "subtypethree", typeId: 3 } },
];

export const server = setupServer(...handlers);
