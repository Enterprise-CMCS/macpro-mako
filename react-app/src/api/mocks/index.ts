export * as mockSubmit from "./submit";
export * as mockTypes from "./types";
