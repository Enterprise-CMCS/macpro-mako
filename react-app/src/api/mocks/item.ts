import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";
import { opensearch, SEATOOL_STATUS } from "shared-types";

type GetItemBody = { id: string };
type ItemTestFields = Pick<
  opensearch.main.Document,
  "id" | "seatoolStatus" | "actionType"
>;

const ID_SEPARATOR = "-";
type IdParamKey = keyof opensearch.main.Document;
// Because getItem(id: string) doesn't allow for easy object mocking,
// to make tests easier, you can add params to the ids your tests use
// and mock specific attributes.
// e.x. existing-approved-actionType=New
const getIdParam = (id: string, key: IdParamKey) =>
  id
    .split(ID_SEPARATOR)
    .find((param: string) => param.includes(key))
    ?.slice(key.length + 1); // + 1 to cover the `=` sign

const handlers = [
  http.post<GetItemBody, GetItemBody>(
    "/item-mock-server",
    async ({ request }) => {
      const { id } = await request.json();
      return id.includes("existing")
        ? HttpResponse.json({
            _id: id,
            _source: {
              id: id,
              seatoolStatus: id.includes("approved")
                ? SEATOOL_STATUS.APPROVED
                : SEATOOL_STATUS.PENDING,
              actionType: getIdParam(id, "actionType") || "New",
            } satisfies ItemTestFields,
          })
        : new HttpResponse(null, { status: 404 });
    },
  ),
  http.post<GetItemBody, GetItemBody>(
    "/item-mock-server-bad",
    async ({ request }) => {
      const { id } = await request.json();
      return id.includes("existing")
        ? HttpResponse.json({
            _id: id,
            _source: {
              id: id,
              seatoolStatus: id.includes("approved")
                ? SEATOOL_STATUS.APPROVED
                : SEATOOL_STATUS.PENDING,
              actionType: getIdParam(id, "actionType") || "New",
            } satisfies ItemTestFields,
          })
        : new HttpResponse(null, { status: 404 });
    },
  ),
];

export const server = setupServer(...handlers);
