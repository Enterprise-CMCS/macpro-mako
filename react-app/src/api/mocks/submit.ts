import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

const handlers = [
  http.put("/upload", async () => {
    return new HttpResponse(null, { status: 200 });
  }),
  http.post("/getUploadUrl", () => {
    return HttpResponse.json(
      {
        url: "/upload",
        key: "test-key",
        bucket: "test-bucket",
      },
      { status: 200 },
    );
  }),
  http.post("/test", () => {
    return HttpResponse.json({ message: "pass" }, { status: 200 });
  }),
];

export const server = setupServer(...handlers);
