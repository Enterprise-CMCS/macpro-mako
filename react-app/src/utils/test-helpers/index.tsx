export * from "./dashboard";
export * from "./render";
export * from "./renderForm";
export * from "./skipCleanup";
export * from "./uploadFiles";
