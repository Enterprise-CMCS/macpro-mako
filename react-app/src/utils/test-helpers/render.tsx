import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import React, { ReactElement } from "react";
import { createMemoryRouter, MemoryRouter, RouterProvider } from "react-router";
import { render } from "@testing-library/react";

export const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
  });

export const queryClientWrapper = ({ children }: { children: ReactElement }) => (
  <QueryClientProvider client={createTestQueryClient()}>{children}</QueryClientProvider>
);

export const memoryRouterWrapper = ({ children }: { children: ReactElement }) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export const renderWithQueryClient = (element: ReactElement) =>
  render(element, {
    wrapper: ({ children }) => (
      <QueryClientProvider client={createTestQueryClient()}>
        <MemoryRouter>{children}</MemoryRouter>,
      </QueryClientProvider>
    ),
  });

export const renderWithMemoryRouter = (
  element: ReactElement,
  ...routing: Parameters<typeof createMemoryRouter>
) => {
  const router = createMemoryRouter(...routing);
  const rendered = render(element, {
    wrapper: () => <RouterProvider router={router} />,
  });
  return {
    router,
    ...rendered,
  };
};

export const renderWithQueryClientAndMemoryRouter = (
  element: ReactElement,
  ...routing: Parameters<typeof createMemoryRouter>
) => {
  const router = createMemoryRouter(...routing);
  const rendered = render(element, {
    wrapper: () => (
      <QueryClientProvider client={createTestQueryClient()}>
        <RouterProvider router={router} />
      </QueryClientProvider>
    ),
  });
  return {
    router,
    ...rendered,
  };
};
