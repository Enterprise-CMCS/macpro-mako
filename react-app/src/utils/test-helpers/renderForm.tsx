import { waitForElementToBeRemoved } from "@testing-library/react";
import React, { ReactElement } from "react";
import items from "mocks/data/items";
import { Authority } from "shared-types";
import { renderWithQueryClient, renderWithQueryClientAndMemoryRouter } from "./render";

export const renderFormAsync = async (form: ReactElement) => {
  const container = await renderWithQueryClient(form);

  // wait for loading
  if (container.queryAllByLabelText("three-dots-loading")?.length > 0) {
    await waitForElementToBeRemoved(() => container.queryAllByLabelText("three-dots-loading"));
  }

  return container;
};

export const renderFormWithPackageSectionAsync = async (
  form: ReactElement,
  id: string,
  authority?: string,
) => {
  const routes = [
    {
      path: "/dashboard",
      element: <h1>dashboard test</h1>,
    },
    {
      path: "/test/:id/:authority",
      element: form,
    },
  ];
  const routeOptions = {
    initialEntries: [
      `/test/${id}/${(authority || items[id]?._source?.authority || "Medicaid SPA") as Authority}`,
    ],
  };

  const container = await renderWithQueryClientAndMemoryRouter(form, routes, routeOptions);

  // wait for loading
  if (container.queryAllByLabelText("three-dots-loading")?.length > 0) {
    await waitForElementToBeRemoved(() => container.queryAllByLabelText("three-dots-loading"));
  }

  return container;
};
