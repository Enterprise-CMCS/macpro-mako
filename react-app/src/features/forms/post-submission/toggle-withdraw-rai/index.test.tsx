import { WITHDRAW_RAI_ITEM_B } from "mocks";
import { describe, expect, it, vi } from "vitest";

import { renderFormWithPackageSectionAsync } from "@/utils/test-helpers/renderForm";

import { DisableWithdrawRaiForm, EnableWithdrawRaiForm } from ".";

vi.mock("react-router", async () => ({
  ...(await vi.importActual<Record<string, unknown>>("react-router")),
  useParams: vi.fn().mockReturnValue({ authority: "1915(b)", id: WITHDRAW_RAI_ITEM_B }),
}));

vi.mock("shared-utils", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...(typeof actual === "object" && actual !== null ? actual : {}),
    isCmsUser: vi.fn().mockReturnValue(true),
  };
});

describe("Toggle Withdraw Rai components", () => {
  it("renders enable withdraw rai correctly", async () => {
    const { asFragment } = await renderFormWithPackageSectionAsync(<DisableWithdrawRaiForm />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders disable withdraw rai correctly", async () => {
    const { asFragment } = await renderFormWithPackageSectionAsync(<EnableWithdrawRaiForm />);

    expect(asFragment()).toMatchSnapshot();
  });
});
