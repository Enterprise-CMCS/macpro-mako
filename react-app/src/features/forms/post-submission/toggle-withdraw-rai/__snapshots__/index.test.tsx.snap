// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toggle Withdraw Rai components > renders disable withdraw rai correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-screen-xl mx-auto px-4 lg:px-8"
      >
        <nav
          aria-label="breadcrumbs for spa or waiver choices"
          class="my-4"
          role="navigation"
        >
          <ul
            class="flex flex-wrap gap-1"
          >
            <li
              class="flex items-center text-sm"
            >
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/dashboard?tab=waivers"
              >
                Dashboard
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/details/1915(b)/VA-2234.R11.02"
              >
                VA-2234.R11.02
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <span
                aria-disabled="true"
                class="whitespace-nowrap"
              >
                Enable Formal RAI Response Withdraw
              </span>
            </li>
          </ul>
        </nav>
        <form
          class="my-6 space-y-8 mx-auto justify-center flex flex-col"
        >
          <section
            class="p-4 border rounded-sm border-cardBorder"
            data-testid="detail-section"
          >
            <h1
              class="text-3xl font-semibold mb-2"
              data-testid="detail-section-title"
            >
              Enable Formal RAI Response Withdraw Details
            </h1>
            <hr
              class="my-6 bg-slate-200"
            />
            <div
              class="gap-8 flex flex-col"
              data-testid="detail-section-child"
            >
              <div>
                <div
                  class="mt-4"
                >
                  Once you submit this form, the most recent Formal RAI Response for this package will be able to be withdrawn by the state
                </div>
              </div>
              <section
                class="flex flex-col mb-8 space-y-8"
              >
                <div>
                  <p
                    class="font-bold"
                  >
                    Waiver Number
                  </p>
                  <p
                    class="text-xl"
                  >
                    VA-2234.R11.02
                  </p>
                </div>
                <div>
                  <p
                    class="font-bold"
                  >
                    Authority
                  </p>
                  <p
                    class="text-xl"
                  >
                    1915(b) Waiver
                  </p>
                </div>
              </section>
            </div>
          </section>
          <section
            class="flex justify-end gap-2 p-4 ml-auto"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
              data-testid="submit-action-form"
              disabled=""
              type="submit"
            >
              Submit
            </button>
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
              data-testid="cancel-action-form"
              type="reset"
            >
              Cancel
            </button>
          </section>
        </form>
        <div
          class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
          role="alert"
        >
          <p
            class="text-lg"
          >
            Do you have questions or need support?
          </p>
          <a
            data-discover="true"
            href="/faq"
            target="_blank"
          >
            <button
              class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
            >
              View FAQs
            </button>
          </a>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-screen-xl mx-auto px-4 lg:px-8"
    >
      <nav
        aria-label="breadcrumbs for spa or waiver choices"
        class="my-4"
        role="navigation"
      >
        <ul
          class="flex flex-wrap gap-1"
        >
          <li
            class="flex items-center text-sm"
          >
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/dashboard?tab=waivers"
            >
              Dashboard
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/details/1915(b)/VA-2234.R11.02"
            >
              VA-2234.R11.02
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <span
              aria-disabled="true"
              class="whitespace-nowrap"
            >
              Enable Formal RAI Response Withdraw
            </span>
          </li>
        </ul>
      </nav>
      <form
        class="my-6 space-y-8 mx-auto justify-center flex flex-col"
      >
        <section
          class="p-4 border rounded-sm border-cardBorder"
          data-testid="detail-section"
        >
          <h1
            class="text-3xl font-semibold mb-2"
            data-testid="detail-section-title"
          >
            Enable Formal RAI Response Withdraw Details
          </h1>
          <hr
            class="my-6 bg-slate-200"
          />
          <div
            class="gap-8 flex flex-col"
            data-testid="detail-section-child"
          >
            <div>
              <div
                class="mt-4"
              >
                Once you submit this form, the most recent Formal RAI Response for this package will be able to be withdrawn by the state
              </div>
            </div>
            <section
              class="flex flex-col mb-8 space-y-8"
            >
              <div>
                <p
                  class="font-bold"
                >
                  Waiver Number
                </p>
                <p
                  class="text-xl"
                >
                  VA-2234.R11.02
                </p>
              </div>
              <div>
                <p
                  class="font-bold"
                >
                  Authority
                </p>
                <p
                  class="text-xl"
                >
                  1915(b) Waiver
                </p>
              </div>
            </section>
          </div>
        </section>
        <section
          class="flex justify-end gap-2 p-4 ml-auto"
        >
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
            data-testid="submit-action-form"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
            data-testid="cancel-action-form"
            type="reset"
          >
            Cancel
          </button>
        </section>
      </form>
      <div
        class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
        role="alert"
      >
        <p
          class="text-lg"
        >
          Do you have questions or need support?
        </p>
        <a
          data-discover="true"
          href="/faq"
          target="_blank"
        >
          <button
            class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
          >
            View FAQs
          </button>
        </a>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggle Withdraw Rai components > renders disable withdraw rai correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-screen-xl mx-auto px-4 lg:px-8"
      >
        <nav
          aria-label="breadcrumbs for spa or waiver choices"
          class="my-4"
          role="navigation"
        >
          <ul
            class="flex flex-wrap gap-1"
          >
            <li
              class="flex items-center text-sm"
            >
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/dashboard?tab=waivers"
              >
                Dashboard
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/details/1915(b)/VA-2234.R11.02"
              >
                VA-2234.R11.02
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <span
                aria-disabled="true"
                class="whitespace-nowrap"
              >
                Disable Formal RAI Response Withdraw
              </span>
            </li>
          </ul>
        </nav>
        <form
          class="my-6 space-y-8 mx-auto justify-center flex flex-col"
        >
          <section
            class="p-4 border rounded-sm border-cardBorder"
            data-testid="detail-section"
          >
            <h1
              class="text-3xl font-semibold mb-2"
              data-testid="detail-section-title"
            >
              Disable Formal RAI Response Withdraw Details
            </h1>
            <hr
              class="my-6 bg-slate-200"
            />
            <div
              class="gap-8 flex flex-col"
              data-testid="detail-section-child"
            >
              <div>
                <div
                  class="mt-4"
                >
                  The state will not be able to withdraw its RAI response. It may take up to a minute for this change to be applied.
                </div>
              </div>
              <section
                class="flex flex-col mb-8 space-y-8"
              >
                <div>
                  <p
                    class="font-bold"
                  >
                    Waiver Number
                  </p>
                  <p
                    class="text-xl"
                  >
                    VA-2234.R11.02
                  </p>
                </div>
                <div>
                  <p
                    class="font-bold"
                  >
                    Authority
                  </p>
                  <p
                    class="text-xl"
                  >
                    1915(b) Waiver
                  </p>
                </div>
              </section>
            </div>
          </section>
          <section
            class="flex justify-end gap-2 p-4 ml-auto"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
              data-testid="submit-action-form"
              disabled=""
              type="submit"
            >
              Submit
            </button>
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
              data-testid="cancel-action-form"
              type="reset"
            >
              Cancel
            </button>
          </section>
        </form>
        <div
          class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
          role="alert"
        >
          <p
            class="text-lg"
          >
            Do you have questions or need support?
          </p>
          <a
            data-discover="true"
            href="/faq"
            target="_blank"
          >
            <button
              class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
            >
              View FAQs
            </button>
          </a>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-screen-xl mx-auto px-4 lg:px-8"
    >
      <nav
        aria-label="breadcrumbs for spa or waiver choices"
        class="my-4"
        role="navigation"
      >
        <ul
          class="flex flex-wrap gap-1"
        >
          <li
            class="flex items-center text-sm"
          >
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/dashboard?tab=waivers"
            >
              Dashboard
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/details/1915(b)/VA-2234.R11.02"
            >
              VA-2234.R11.02
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <span
              aria-disabled="true"
              class="whitespace-nowrap"
            >
              Disable Formal RAI Response Withdraw
            </span>
          </li>
        </ul>
      </nav>
      <form
        class="my-6 space-y-8 mx-auto justify-center flex flex-col"
      >
        <section
          class="p-4 border rounded-sm border-cardBorder"
          data-testid="detail-section"
        >
          <h1
            class="text-3xl font-semibold mb-2"
            data-testid="detail-section-title"
          >
            Disable Formal RAI Response Withdraw Details
          </h1>
          <hr
            class="my-6 bg-slate-200"
          />
          <div
            class="gap-8 flex flex-col"
            data-testid="detail-section-child"
          >
            <div>
              <div
                class="mt-4"
              >
                The state will not be able to withdraw its RAI response. It may take up to a minute for this change to be applied.
              </div>
            </div>
            <section
              class="flex flex-col mb-8 space-y-8"
            >
              <div>
                <p
                  class="font-bold"
                >
                  Waiver Number
                </p>
                <p
                  class="text-xl"
                >
                  VA-2234.R11.02
                </p>
              </div>
              <div>
                <p
                  class="font-bold"
                >
                  Authority
                </p>
                <p
                  class="text-xl"
                >
                  1915(b) Waiver
                </p>
              </div>
            </section>
          </div>
        </section>
        <section
          class="flex justify-end gap-2 p-4 ml-auto"
        >
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
            data-testid="submit-action-form"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
            data-testid="cancel-action-form"
            type="reset"
          >
            Cancel
          </button>
        </section>
      </form>
      <div
        class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
        role="alert"
      >
        <p
          class="text-lg"
        >
          Do you have questions or need support?
        </p>
        <a
          data-discover="true"
          href="/faq"
          target="_blank"
        >
          <button
            class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
          >
            View FAQs
          </button>
        </a>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
